astropy/tests/helper.py

```diff
@@ -200,3 +201,68 @@ def assert_quantity_allclose(actual, desired, rtol=1.0e-7, atol=None, **kwargs):
     np.testing.assert_allclose(
         *_unquantify_allclose_arguments(actual, desired, rtol, atol), **kwargs
     )
+
+
+def assert_objects_equal(actual, desired, /, *, attrs=None, compare_class=True):
+    """Compare ndarray objects the attribute level.
+
+    Parameters
+    ----------
+    actual and desired: np.ndarray (positional only)
+
+    attrs: list[str] (optional, keyword only)
+       a list of attributes to compare in addition to the default list:
+       ['info.name', 'info.format', 'info.unit',
+        'info.description', 'info.meta', 'info.dtype']
+
+    compare_class: bool (default: False, keyword only)
+      whether to compare object classes via the equality operator (==)
```

Comment:
I would just write: "Whether to check that the two objects are instances of the same class"